server/src/main/java/org/opensearch/plugins/SecureTransportSettingsProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.http.HttpServerTransport;
+import org.opensearch.transport.TcpTransport;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+
+import java.util.Optional;
+
+/**
+ * A provider for security related settings for transports.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface SecureTransportSettingsProvider {
+    /**
+     * An exception handler for errors that might happen while secure transport handle the requests.
+     *
+     * @see <a href="https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/SslExceptionHandler.java">SslExceptionHandler</a>
+     *
+     * @opensearch.experimental
+     */
+    @ExperimentalApi
+    @FunctionalInterface
+    interface ServerExceptionHandler {
+        static ServerExceptionHandler NOOP = t -> {};
+
+        /**
+         * Handler for errors happening during the server side processing of the requests
+         * @param t the error
+         */
+        void onError(Throwable t);
+    }
+
+    /**
+     * If supported, builds the {@link ServerExceptionHandler} instance for {@link HttpServerTransport} instance
+     * @param settings settings
+     * @param transport {@link HttpServerTransport} instance
+     * @return if supported, builds the {@link ServerExceptionHandler} instance
+     */
+    Optional<ServerExceptionHandler> buildHttpServerExceptionHandler(Settings settings, HttpServerTransport transport);
```

Comment:
Why return an optional handler?  I'd rather enforce a handler is required explicitly, even if it is the NOOP version you've included above.

Reply:
Looks like this Optional parameters is a pattern for this interface, what is the use case for an interface that isn't requiring a structured functionality?

Reply:
The are transport plugins that only provide HTTP engine and not transport engine (`reactor-netty` is one of those) and vice versa, so every part is essentially optional

Reply:
Its got a 'code smell' to it.  What about multiple interfaces instead of a single 'catch' all?

Reply:
> I think starting with minimal APIs gives us a good starting point to build on top,  

That could be option actually, it is a bit hard to say with one impl (netty4), but adding implementing the secure transport for `reactor-netty` and `nio` would actually give us a better understand if we have to split this interface or not, thank for insight